app/keepers/keepers.go

```diff
@@ -316,6 +319,16 @@ func (appKeepers *AppKeepers) InitNormalKeepers(
 	)
 	appKeepers.TokenFactoryKeeper = &tokenFactoryKeeper
 
+	validatorPreferenceKeeper := validatorpreference.NewKeeper(
+		appKeepers.keys[validatorpreferencetypes.StoreKey],
+		appKeepers.GetSubspace(validatorpreferencetypes.ModuleName),
+		appKeepers.StakingKeeper,
+		appKeepers.BankKeeper,
+		appKeepers.DistrKeeper,
+	)
+
+	appKeepers.ValidatorPreferenceKeeper = &validatorPreferenceKeeper
```

Comment:
Why is a reference to the keeper needed?

Reply:
that was a part of wiring up the keeper

Reply:
Yes, but why is a reference necessary? Typically keepers are non-mutative -- there are a few exceptions like staking where we need to set hooks after the fact.